vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -109,6 +135,32 @@ def parse_pod(pod)
       new_result
     end
 
+    def parse_endpoint(entity)
+      new_result = parse_base_item(entity)
+      new_result[:container_groups] = []
+
+      # remove once kubernetes api v1beta3/v1.0 will be stable, see:
+      # https://github.com/GoogleCloudPlatform/kubernetes/commit/8ae20382
+      if !entity.respond_to?(:subsets) || entity.subsets.nil?
+        $log.warn("your kubernetes endpoints are not providing subsets, " \
+                  "please upgrade (see commit 8ae20382")
```

Comment:
- I think you are missing an extra paren inside the string
- This warning references a commit SHA, but out of context, in the log, that commit means nothing.  You might want to just put the full URL into the message.